Make ProjectReference/Project metadata tag optional

matherm-aboehm opened the pull request:
Fixes wixtoolset/issues#4186

matherm-aboehm committed:
Make ProjectReference/Project metadata tag optional

Fixes wixtoolset/issues#4186

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/signed)](https://cla.dotnetfoundation.org/wixtoolset/VisualStudioExtension?pullRequest=15) <br/>All CLA requirements met.

rseanhall commented:
This would appear to fix https://github.com/wixtoolset/issues/issues/3889 as well. Have you tested this solution?

matherm-aboehm committed:
Fix fetch of VisualStudioVersion for VS2017 and VS2019

The registry key is not available in VS2017 and VS2019, but there is instead a new COM API to retrieve the version string.

Includes the source file which declares the COM interface as symbolic link to the file in submodule project-system (repo url: https://github.com/dotnet/project-system.git).

matherm-aboehm committed:
Load the correct version of Microsoft.VisualStudio.VCProjectEngine

Use VisualStudioVersion to load the correct version of Microsoft.VisualStudio.VCProjectEngine assembly.

Fixes wixtoolset/issues#3889

matherm-aboehm commented:
Yes I always test my changes before I submit them.
You are right, this will also fix wixtoolset/issues#3889, though it was not intended, it's a side effect for what is required to fix  wixtoolset/issues#4186.
I will update this PR to include a safer version for the fix of wixtoolset/issues#3889, which uses the actual Visual Studio version to load the right assembly.

matherm-aboehm committed:
Fix AppVeyor config for submodules and symbolic links

matherm-aboehm committed:
Use old syntax to build on VS2015

matherm-aboehm committed:
Add IVsSolutionLoadEvents with virtual stubs to SolutionListener

matherm-aboehm committed:
Implement ISupportInitializeNotification for ProjectReferenceNode

To support deferred project loading scenarios, ProjectReferenceNode needs a mechanism which finishes initialization after constructor was called that ISupportInitializeNotification offers. EndInit() is called by SolutionListenerForProjectReferenceUpdate.OnAfterOpenProject() if initialization wasn't finished in constructor.

Also fix spelling of some comments and debug messages.

robmen commented:
Interesting proposal but the submodule dependency opens this project up for external breakage (especially not worth the risk to grab a single file). For now, closing the stale PR.
